test/extended/operators/check_crd_status.go

```diff
@@ -0,0 +1,121 @@
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	"k8s.io/kube-openapi/pkg/util/sets"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// checkCRDs returns a list of names of CRDs that don't have subresource.status
+// For now, it ignores the ones that are currently failing.
+func checkCRDs(crdItemList []apiextensionsv1.CustomResourceDefinition) []string {
+
+	// These 23 CRDs that, at the time this test was written, do not have subresource.status.
+	// These can be skipped for now but we don't want the number to increase.
+	// The 23 CRDs should be tidied up over time.
+	//
+	exceptionsList := sets.NewString(
+		"builds.config.openshift.io",
+		"clusternetworks.network.openshift.io",
+		"consoleclidownloads.console.openshift.io",
+		"consoleexternalloglinks.console.openshift.io",
+		"consolelinks.console.openshift.io",
+		"consolenotifications.console.openshift.io",
+		"consoleplugins.console.openshift.io",
+		"consolequickstarts.console.openshift.io",
+		"consoleyamlsamples.console.openshift.io",
+		"egressnetworkpolicies.network.openshift.io",
+		"hostsubnets.network.openshift.io",
+		"imagecontentpolicies.config.openshift.io",
+		"imagecontentsourcepolicies.operator.openshift.io",
+		"machineconfigs.machineconfiguration.openshift.io",
+		"netnamespaces.network.openshift.io",
+		"networks.config.openshift.io",
+		"networks.operator.openshift.io",
+		"operatorpkis.network.operator.openshift.io",
+		"profiles.tuned.openshift.io",
+		"rangeallocations.security.internal.openshift.io",
+		"rolebindingrestrictions.authorization.openshift.io",
+		"securitycontextconstraints.security.openshift.io",
+		"tuneds.tuned.openshift.io",
+	)
+
+	failures := []string{}
+	for _, crdItem := range crdItemList {
+
+		// This test is interested only in CRDs that end with "openshift.io".
+		if !strings.HasSuffix(crdItem.ObjectMeta.Name, "openshift.io") {
+			continue
+		}
+
+		crdName := crdItem.ObjectMeta.Name
+
+		// Skip CRDs in the exceptions list for now.
+		if exceptionsList.Has(crdName) {
+			continue
+		}
+
+		// Iterate through all versions of the CustomResourceDefinition Spec; if any version has a status element
+		// defined in the schema, it passes.
+		foundStatusInSchema := false
+		for i := 0; i < len(crdItem.Spec.Versions); i++ {
+			if _, ok := crdItem.Spec.Versions[i].Schema.OpenAPIV3Schema.Properties["status"]; ok {
+				if crdItem.Spec.Versions[i].Subresources != nil && crdItem.Spec.Versions[i].Subresources.Status != nil {
+					foundStatusInSchema = true
+					break
+				}
+			}
+		}
+		if !foundStatusInSchema {
+			failures = append(failures, fmt.Sprintf("CRD %s is missing subresources.status", crdName))
```

Comment:
they aren't required to have status.  Some objects don't.  But if they have status in the schema, they must have a status subresource